### PR TITLE
EPMRPP-104670 || Add selected test cases to test plan, add error handlings, wire up hide added toggle

### PR DESCRIPTION
**Pull request by yelyzavetakhokhlova**
## PR Checklist

* [x] Have you verified that the PR is pointing to the correct target branch? (`develop` for features/bugfixes, other if mentioned in the task)
* [x] Have you verified that your branch is consistent with the target branch and has no conflicts? (if not, make a rebase under the target branch)
* [x] Have you checked that everything works within the branch according to the task description and tested it locally?
* [x] Have you run the linter (`npm run lint`) prior to submission? Enable the git hook on commit in your IDE to run it and format the code automatically.
* [ ] Have you run the tests locally and added/updated them if needed?
* [ ] Have you checked that app can be built (`npm run build`)?
* [ ] Have you checked that no new circular dependencies appreared with your changes? (the webpack plugin reports circular dependencies within the `dev` npm script)
* [ ] Have you made sure that all the necessary pipelines has been successfully completed?
* [ ] If the task requires translations to be updated, have you done this by running the `manage:translations` script?
* [ ] Have you added the link to the PR in the Jira ticket comments?
* [ ] Have you checked and implemented role-based permissions? (if the feature requires different behavior or visibility for different user roles)

## Visuals

https://github.com/user-attachments/assets/b3331a58-aec8-453f-bb69-ecfbbf9ad544

## Summary by CodeRabbit

* **New Features**
  * Async add-to-test-plan with progress state and confirmation; empty-plan button opens the test library.
  * Prefetches test-plan test-case counts per folder to inform UI.

* **UI Improvements**
  * Option to hide test cases already in the selected test plan; folder tree omits folders with all tests added.
  * Shows loader while prefetching; disables actions while adding.
* **Messages**
  * Added success notification and “All test cases are already in test plan.”

### CODE REVIEW

File: app/src/pages/inside/common/testLibrarySidePanel/testLibrarySidePanel.tsx [hunk 1]
```diff
@@ -83,10 +84,14 @@ export const TestLibrarySidePanel = ({
         onClearSelection={clearSelection}
       />
       <div className={cx('test-library-panel__footer-buttons')}>
-        <Button variant="ghost" onClick={addToTestPlan} disabled={selectionCount === 0}>
+        <Button
+          variant="ghost"
+          disabled={selectionCount === 0 || isAddingToTestPlan}
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/common/testLibrarySidePanel/testLibrarySidePanel.tsx]: Let’s create variable/function for it - see that it is used in place below

File: app/src/pages/inside/common/testLibrarySidePanel/testLibrarySidePanel.tsx [hunk 2]
```diff
@@ -28,26 +27,30 @@ import { useTestLibraryPanel } from './hooks/useTestLibraryPanel';
 import { messages } from './messages';
 
 import styles from './testLibrarySidePanel.scss';
+import { useState } from 'react';
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/common/testLibrarySidePanel/testLibrarySidePanel.tsx]: Order

**Review by ViktorSoroka07 — APPROVED** [review]
